dev/wlp.lib.extract/src/wlp/lib/extract/SelfExtractor.java

```diff
@@ -518,85 +518,140 @@ public ReturnCode installMissingRequiredFeatures(File outputDir, ReturnCode rc)
      * matches method must return either {@link ProductMatch#NOT_APPLICABLE} or {@link ProductMatch#MATCHED} for every properties file for this method to return
      * {@link ReturnCode#OK}.
      *
-     * @param outputDir Where the product is being installed to
+     * @param outputDir      Where the product is being installed to
      * @param productMatches The list of {@link ProductMatch} objects that need to be satisfied to the current install
      * @return A {@link ReturnCode} indicating if this was successful or not
      */
     public static ReturnCode validateProductMatches(File outputDir, List productMatches) {
-        boolean dirExists;
-        dirExists = outputDir.exists();
-        if (dirExists) {
-            File f = new File(outputDir, "lib/versions");
-            File[] files = f.listFiles();
+        List<Properties> props_list = getLibertyProperties(outputDir);
+        return validateProperties(productMatches, props_list);
+    }
+
+    /**
+     * @param productMatches
+     * @param rc
+     * @param props_list
+     * @return
+     */
+    protected static ReturnCode validateProperties(List productMatches, List<Properties> props_list) {
+        ReturnCode rc = ReturnCode.OK;
+        if (props_list.isEmpty()) {
+            rc = new ReturnCode(ReturnCode.BAD_OUTPUT);
+        } else {
+            for (int i = 0; i < props_list.size() && isReturnCodeOK(rc); i++) {
+                rc = matchLibertyProperties(productMatches, props_list.get(i));
+            }
+        }
+        return rc;
+    }
+
+    /**
+     * TODO move this to ReturnCode class
```

Comment:
remove this TODO

Reply:
optionally, move this check into ReturnCode or keep it here and remove the TODO.